boa/src/syntax/parser/function_class/mod.rs

```diff
@@ -0,0 +1,18 @@
+//! Functions and classes parsing.
```

Comment:
Why is there a `function_class` module? Shouldn't it just be `function_declaration`?

Furthermore shouldn't  `function_declaration` be in a module `hoistable_declaration` in `declaration` module?

[Declaration spec][spec]

[spec]: https://tc39.es/ecma262/#prod-Declaration